android/java/org/chromium/chrome/browser/util/TabUtils.java

```diff
@@ -364,4 +367,28 @@ public static int getTransition(Tab tab) {
         }
         return 0;
     }
+
+    /**
+     * Returns {@code true} when current tab URL matches the URL of a YouTube video. Current tab or
+     * current tab URL might be {@code null} if the model is not initialized yet.
+     *
+     * @param currentTab Current tab holding the URL to check.
+     * @return {@code true} if the current tab URL matches the URL of a YouTube video, {@code false}
+     *     otherwise.
+     */
+    public static boolean isYouTubeVideo(@Nullable final Tab currentTab) {
```

Comment:
this should not be in a generic `TabUtils` class and probably shouldn't be in java at all, but also this kind of thing needs tests

Reply:
also this is duplicating another check in the cpp code which is also bad. All of these checks should move into the cpp code and have unit tests